relayer/chain/ethereum/listener.go

```diff
@@ -117,27 +125,31 @@ func (li *Listener) pollEventsAndHeaders(
 			}
 
 			finalizedBlockNumber := gethheader.Number.Uint64() - descendantsUntilFinal
-			var events []*outbound.ContractMessage
+			var events []*etypes.Log
+
+			filterOptions := bind.FilterOpts{Start: finalizedBlockNumber, End: &finalizedBlockNumber, Context: ctx}
 
-			for _, channelContract := range li.contracts {
-				channelEvents, err := li.queryEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
-				if err != nil {
-					li.log.WithError(err).Error("Failure fetching event logs")
-				}
+			basicEvents, err := li.queryBasicEvents(li.basicOutboundChannel, &filterOptions)
```

Comment:
Just want to point out that we are reading events from the two channels serially. Is there a potential scenario in which reading basic channel events could hinder the performance of the incentivized one?
Probably not a problem, or at least not yet, but it caught my attention.

Reply:
yeah is potentially an issue. Though at some point we will probably modify the relayer so that it works with only one channel, either the basic one, or the incentivized one.

Reply:
yep @blablacio is working on a more configurable relayer atm already